src/lib/hosts-file.ts

```diff
@@ -0,0 +1,177 @@
+import fs from 'fs';
+import { platform } from 'os';
+import { promisify } from 'util';
+import sudo from 'sudo-prompt';
+
+const readFile = promisify( fs.readFile );
+const writeFile = promisify( fs.writeFile );
+const sudoExec = promisify( sudo.exec );
+
+// Host file paths for different operating systems
+const HOST_FILES: Record< string, string > = {
+	win32: 'C:\\Windows\\System32\\drivers\\etc\\hosts',
+	darwin: '/etc/hosts',
+	linux: '/etc/hosts',
+};
+
+// Get the appropriate hosts file path for the current platform
+const getHostsFilePath = (): string => {
+	const currentPlatform = platform();
+	const hostsPath = HOST_FILES[ currentPlatform ];
+
+	if ( ! hostsPath ) {
+		throw new Error( `Unsupported platform: ${ currentPlatform }` );
+	}
+
+	return hostsPath;
+};
+
+/**
+ * Reads the hosts file content
+ */
+export const readHostsFile = async (): Promise< string > => {
+	try {
+		const hostsPath = getHostsFilePath();
+		const content = await readFile( hostsPath, 'utf8' );
+		return content;
+	} catch ( error ) {
+		console.error( 'Error reading hosts file:', error );
+		throw error;
+	}
+};
+
+/**
+ * Writes content to the hosts file (requires elevated permissions)
+ */
+export const writeHostsFile = async ( content: string ): Promise< void > => {
+	const hostsPath = getHostsFilePath();
+	try {
+		const tempPath = '/tmp/wp-studio-hosts';
+		await writeFile( tempPath, content );
+		// @ts-expect-error promisify doesn't seem typed properly.
+		await sudoExec( `cat ${ tempPath } > ${ hostsPath }`, {
```

Comment:
It asks for permission each time I add a new site. Is there a way to save it somehow, at least for the current Studio session?

Reply:
Correct, now it asks for permissions only when I add a new site, and I choose to use a custom domain. It doesn't ask for permission when I start or stop the site afterward. It seems acceptable, and we could improve it in the future if possible.